### ISSUE TITLE
Potential Issue: Residue Masking Only Applied During Sampling, Not Inference

### ISSUE THREAD

**Issue by valentin-noske** [oxpig/AntiFold#23]
Hello,

I have a question regarding the `get_pdb_logits` function. From my understanding of the code, residue masking is applied only during the sampling process (`sample_from_df_logits_H`, `sample_from_df_logits_HL`), not during inference with the model. This would mean that, during inference, the model already has access to the entire sequence before any sampling occurs.

I might be mistaken, but I wanted to raise this point in case it is unintended behavior.

**Comment by Magnushhoie** [comment]
Hi valentin, AntiFold is trained like other inverse folding models to recover masked residues, but the default inference here is a single forward pass without masking, where surrounding residues may inform the one to be predicted. This is mainly for speed, and since sequence information is almost always available in real-world scenarios.

However you can manually set a mask by applying mask tokens to the positions to be predicted, or adapt ESM-IF1s sample method to iteratively mask and predict one token at a time: https://github.com/facebookresearch/esm/blob/2b369911bb5b4b0dda914521b9475cad1656b2ac/esm/inverse_folding/gvp_transformer.py#L88

We never got around to adding these features, but I'm happy to merge a pull request related to this.

**Comment by valentin-noske** [comment]
Thanks a lot for the reply. :) 

I definitely implemented a version using the ESM-IF sampling method. I'll try to create a pull request, maybe early next month, once I have time to make it look more appealing and think about where to add this feature.

Maybe, it would be helpful to update the README.md to make it clearer that this method does not mask the sequence of the structure, as this distinction can otherwise be a little confusing.
